### PR TITLE
Update jruby version to `9.3.4.0`

### PR DESCRIPTION
**Pull request by robbavey**
This commit updates the version of jruby used in Logstash to `9.3.4.0`.

* Updates the references of `jruby` from `9.2.20.1` to `9.3.4.0`
* Updates references/locations of ruby from `2.5.0` to `2.6.0`
* Updates java imports including `org.logstash.util` to be quoted
  * Without quoting the name of the import, the following error is observed in tests:
  * `java.lang.NoClassDefFoundError: org/logstash/Util (wrong name: org/logstash/util)`
  * Maybe an instance of https://github.com/jruby/jruby/issues/4861
* Adds a monkey patch to `require` to resolve compatibility issue between latest `jruby` and `polyglot` gem 
  * The addition of https://github.com/jruby/jruby/pull/7145 to disallow circular
causes, will throw when `polyglot` is thrown into the mix, and stop logstash from
starting and building - any gems that use an exception to determine whether or not
to load the native gem, will trigger the code added in that commit.
  * This commit adds a monkey patch of `require` to rollback the circular cause exception
back to the original cause.
* Removes the use of the deprecated `JavaClass`
* Adds additional `require time` in `generate_build_metadata`
* Rewrites a test helper to avoid potentially calling `~>` on `FalseClass`

## Release notes
Update jruby dependencies to latest `9.3.4.0` version.

## Why is it important/What is the impact to the user?

jruby-9.3.4.0 includes numerous bug fixes, and is required in order to be able to allow logstash to run natively on m1 macs.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## How to test this PR locally

Run logstash, test with bundled JDK, JDK17. Ideally, we would like to see this run on Windows, running native Java plugins

### CODE REVIEW

**Comment by robbavey** [comment]
Full CI run: https://logstash-ci.elastic.co/job/elastic-logstash-main/180/

**Comment by robbavey** [comment]
JDK CI run: https://logstash-ci.elastic.co/job/elastic-logstash-main-matrix-jdk/48/

Note that Windows CI test will fail due to #14075

File: logstash-core/src/test/java/org/logstash/config/ir/RubyEnvTestCase.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
 import java.nio.file.Path;
 
 import org.jruby.RubyHash;
+import org.jruby.runtime.builtin.IRubyObject;
```
Comment (andsel) [inline logstash-core/src/test/java/org/logstash/config/ir/RubyEnvTestCase.java:26]: The import of `IRubyObject` is unused

**Review by mashhurs — APPROVED** [review]
🥇

**Pull request merged**